### DISCUSSION TITLE
Restart a running container but got a new one

### DISCUSSION THREAD

**Discussion by jiayu1016** [containerd/nerdctl#2022] [General]
There is a running container 0a6466e659be, I want restart the container by :

nerdctl -n k8s.io container restart 0a6466e659be

But I get a failed response:
FATA[0010] failed to start shim: mkdir /run/containerd/io.containerd.runtime.v2.task/k8s.io/0a6466e659becf2e05a56e369bed0ad20f242b85307fd79c26408dc605a83818: file exists: unknown

And a new container 3b5a5c04108d was created

version:
Client:
 Version:       v1.2.0
 OS/Arch:       linux/arm64
 Git commit:    5aee2f754a2f46d4c8ccbae25d56b43668ac4b62
 buildctl:
  Version:

Server:
 containerd:
  Version:      1.6.6
  GitCommit:    10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:      1.1.4
  GitCommit:    5fd4c4d144137e991c4acebb2146ab1483a97925

**Comment by fahedouch** [comment]
it seems like the task is not deleted/cleaned properly. Can you remove this container and test on new proper container ?

**Reply by jiayu1016** [reply]
The same issue as https://github.com/containerd/nerdctl/issues/1274
nerdctl can restart the container created by itself
But restart the container in k8s.io namespace will got the error
I suspect kubelet and nerdctl restarting the container at the same time cause the issue
